torchtext/datasets/text_classification.py

```diff
@@ -27,26 +27,46 @@
 }
 
 
-# TODO: Replicate below
-#  tr '[:upper:]' '[:lower:]' | sed -e 's/^/__label__/g' | \
-#    sed -e "s/'/ ' /g" -e 's/"//g' -e 's/\./ \. /g' -e 's/<br \/>/ /g' \
-#        -e 's/,/ , /g' -e 's/(/ ( /g' -e 's/)/ ) /g' -e 's/\!/ \! /g' \
-#        -e 's/\?/ \? /g' -e 's/\;/ /g' -e 's/\:/ /g' | tr -s " "
-_normalize_pattern_re = re.compile(r'[\W_]+')
+_normalize_pattern_re = [re.compile(r'\''), re.compile(r'\"'),
+                         re.compile(r'\.'), re.compile(r'<br \/>'),
+                         re.compile(r','), re.compile(r'\('),
+                         re.compile(r'\)'), re.compile(r'\!'),
+                         re.compile(r'\?'), re.compile(r'\;'),
+                         re.compile(r'\:'), re.compile(' +')]
+
+replaced_string = [' \'  ', '',
+                   ' . ', ' ',
+                   ' , ', ' ( ',
+                   ' ) ', ' ! ',
+                   ' ? ', ' ',
+                   ' ', ' ']
 
 
 def text_normalize(line):
     """
     Basic normalization for a line of text.
     Normalization includes
     - lowercasing
-    - replacing all non-alphanumeric characters with whitespace
+    - complete some basic text normalization for English words as follows:
+        add spaces before and after '\''
+        remove '\"',
+        add spaces before and after '.'
+        replace '<br \/>'with single space
+        add spaces before and after ','
+        add spaces before and after '('
+        add spaces before and after ')'
+        add spaces before and after '!'
+        add spaces before and after '?'
+        replace ';' multiple spaces with single space
```

Comment:
What does this mean?

Reply:
typo. replace multiple spaces with single space

Reply:
But doesn't that only happen in the end? It looks like the corresponding re only replaces ';' with a single space

Reply:
it replaces ';' with a single space. At the very end, multiple spaces will be combined together.